OADP-4058: Adding structure for OADP and 3scale user story (#85994)
openshift/openshift-docs: modules/backing-up-the-backend-redis-database.adoc

:_mod-docs-content-type: PROCEDURE

//included in backing-up-and-restoring-3scale-using-oadp.adoc assembly

[id="backing-up-the-backend-redis-database_{context}"]
= Backing up the backend Redis database

> You can backup the Redis database by adding the annotations mentioned in this procedure and by adding the resources using the `includedResources` parameter, which must be backed up.

Reviewer: ```suggestion
You can back up the Redis database by adding the annotations mentioned in the following procedure and by adding the resources by using the `includedResources` parameter, which must be backed up.
```

By default, 3scale Redis uses AOF persistence. 

.Prerequisites

. You backed up the Operator.
. You backed up the mysql database.
. Ensure that the Redis queues have been drained before performing backups.

.Procedure

> . Edit the annotations on the `backend-redis` deployment by running the following command:

Reviewer: ```suggestion
. Edit the annotations on the `backend-redis` deployment:
```
" by running the following command" can really be dropped.

Author: I agree but it was suggested in the OCP review

+
[source, terminal]
----
$ oc edit deployment backend-redis -n threescale
----

. Add the following annotations for `backend-redis`:
+
[source,yaml]
----
annotations:
post.hook.backup.velero.io/command: >-
         ["/bin/bash", "-c", "redis-cli CONFIG SET auto-aof-rewrite-percentage
         100"]
       pre.hook.backup.velero.io/command: >-
         ["/bin/bash", "-c", "redis-cli CONFIG SET auto-aof-rewrite-percentage
         0"]
----

. Create the following YAML file to backup the Redis database:
+
.Example redis-backup.yaml:
+
[source,yaml]
----
apiVersion: velero.io/v1
kind: Backup
metadata:
  name: redis-backup
  namespace: openshift-adp
spec:
  csiSnapshotTimeout: 10m0s
  defaultVolumesToFsBackup: true
  includedNamespaces:
  - threescale
  includedResources:
  - deployment
  - pods
  - replicationcontrollers
  - persistentvolumes
  - persistentvolumeclaims
  itemOperationTimeout: 1h0m0s
  labelSelector:
    matchLabels:
      app: 3scale-api-management
      threescale_component: backend
      threescale_component_element: redis
  snapshotMoveData: false
  snapshotVolumes: false
  ttl: 720h0m0s
----

> . Backup the Redis database using the following command:

Reviewer: ```suggestion
. Back up the Redis database:
```
Same  here, no need for "by running the following command"

+
[source,terminal]
----
$ oc get backup redis-backup -o yaml
----

.Verification

> * Verify if the Redis backup is completed: 

Reviewer: ```suggestion
* Verify that the Redis backup is completed: 
```
